src/client.rs

```diff
@@ -250,6 +253,36 @@ impl Client {
     {
         Api::namespaced(self.client.clone(), namespace)
     }
+
+    /// Waits until given resource with metadata is recognized as applied by Kubernetes API
```

Comment:
Only when I saw the usage in crd.rs did I understand how this is to be used.
When you say `given resource` the way it is _given_  is via a `field_selector` in the `ListParams`.

An example would be great here or at least a quick sentence about that.

Your method is definitely the most flexible solution so I'm happy to merge as is.
For convenience we could add a method that wraps this because I assume watching by name will be the most common scenario. But we can do that once we needed.

Reply:
This is great observation, thank you. I reworded the whole documentation and added a usage example that shows full usage, including timeout.

Reply:
Regarding the convenience method - I'll happily add it if required. I usually add stuff once needed, so I'd just do it like that personally.